python/dgl/sparse/elementwise_op_sp.py

```diff
@@ -42,24 +42,19 @@ def sp_add(A: SparseMatrix, B: Union[DiagMatrix, SparseMatrix]) -> SparseMatrix:
     values=tensor([40, 20, 60]),
     shape=(3, 4), nnz=3)
     """
-    if isinstance(B, DiagMatrix):
-        B = B.as_sparse()
-    if isinstance(A, SparseMatrix) and isinstance(B, SparseMatrix):
-        return spsp_add(A, B)
-    raise RuntimeError(
-        "Elementwise addition between {} and {} is not "
-        "supported.".format(type(A), type(B))
-    )
+    # Python falls back to B.__radd__ then TypeError when NotImplemented is
+    # returned.
+    return spsp_add(A, B) if isinstance(B, SparseMatrix) else NotImplemented
```

Comment:
Wait a moment. Does this mean that `A + I` is no longer supported?

Reply:
When Python tries `A.__add__(I)` that returns `NotImplemented`, Python will then try `I.__radd__(A)`.